grug/math/src/number.rs

```diff
@@ -125,11 +101,15 @@ where
     }
 
     fn checked_mul(self, other: Self) -> MathResult<Self> {
-        self.0
-            .checked_full_mul(*other.numerator())?
-            .checked_div(Self::DECIMAL_FRACTION.into_next())?
-            .checked_into_prev()
-            .map(Self)
+        let clos = || {
+            self.0
+                .checked_full_mul(*other.numerator())?
+                .checked_div(Self::DECIMAL_FRACTION.into_next())?
+                .checked_into_prev()
+                .map(Self)
+        };
+
+        clos().map_err(|_| MathError::overflow_mul(self, other))
```

Comment:
It's unnecessary to define the `clos`? You can simply do:

```rust
self.0
    .checked_full_mul(*other.numerator())?
    .checked_div(Self::DECIMAL_FRACTION.into_next())?
    .checked_into_prev()
    .map(Self)
    .map_err(|_| MathError::overflow_mul(self, other))
```

Reply:
It's because if the mul overflow, another error another error is raised. In this way any error raised is converted into OverflowPow

Reply:
I see, you're right. Still you can do:

```rust
{
    // ...
}
.map_err(|_| MathError::overflow_mul(self, other))
```